src/wp-includes/class-wp-block-patterns-registry.php

```diff
@@ -184,19 +188,32 @@ public function get_registered( $pattern_name ) {
 	 *                 and per style.
 	 */
 	public function get_all_registered( $outside_init_only = false ) {
-		$patterns = array_values(
-			$outside_init_only
+		$patterns = $outside_init_only
 				? $this->registered_patterns_outside_init
-				: $this->registered_patterns
-		);
+				: $this->registered_patterns;
+
+		foreach ( $patterns as $pattern_name => $pattern ) {
+			if ( ! $pattern['parsed'] ) {
+				$pattern['content']        = $this->get_parse_blocks( $pattern );
+				$pattern['parsed']         = true;
+				$patterns[ $pattern_name ] = $pattern;
+			}
+		}
 
-		foreach ( $patterns as $index => $pattern ) {
-			$blocks                        = parse_blocks( $pattern['content'] );
-			$before_block_visitor          = make_before_block_visitor( $pattern );
-			$after_block_visitor           = make_after_block_visitor( $pattern );
-			$patterns[ $index ]['content'] = traverse_and_serialize_blocks( $blocks, $before_block_visitor, $after_block_visitor );
+		if ( $outside_init_only ) {
+			$this->registered_patterns_outside_init = $patterns;
+		} else {
+			$this->registered_patterns = $patterns;
 		}
-		return $patterns;
+
+		return array_values( $patterns );
+	}
+
+	protected function get_parse_blocks( $pattern ) {
```

Comment:
Nitpick, we might want a different name for the method, as it not only parses the content but also (conditionally) inserts hooked blocks, and returns re-serialized markup.

(Per https://github.com/WordPress/wordpress-develop/pull/5421#issuecomment-1752811474, we might just use [`pepare_content`](https://github.com/WordPress/wordpress-develop/pull/5399/files#diff-119add4c8563b7b812d6ee6496a8416e25ecd0ef945310478d8344c0bbcdee93R162) from https://github.com/WordPress/wordpress-develop/pull/5399 as name.)